webapp/src/actions.ts

```diff
@@ -409,20 +416,30 @@ export const removeIncomingCallNotification = (callID: string): ActionFunc => {
 export const ringForCall = (callID: string, sound: string) => {
     return async (dispatch: DispatchFunc) => {
         notificationSounds?.ring(sound);
+
+        // window.e2eNotificationsSoundedAt is added when running the e2e tests
+        if (window.e2eNotificationsSoundedAt) {
+            window.e2eNotificationsSoundedAt.push(Date.now());
+        }
```

Comment:
I am sure you put some serious thought into this but can't help but feeling a bit unsatisfied by the need to tinker with application code purely to support tests.

Reply:
That's another case of me spending too long trying to spy the browser and giving up (notifications, sound start, sound stop), plus we'd have to add this sort of thing to the notifications rejected code anyway (this is the test that would have caught the upstream regression we dealt with a few weeks ago). 
It was unsatisfying at first for me too... But then I realized: it's simple, it always works, and it's less prone to breakage.

Reply:
This is a bit like mocking I guess. We are essentially putting a limit to the reach of the test. I suppose this would make them pass even if the webapp never actually sent the browser notification.

Reply:
If by `even if the webapp never actually sent the browser notification` you mean the physical desktop notification, yep -- and that's what we would have mocked on the browser anyway (the fact that the `new Notification` was called, not that the test "desktop" showed a notification...). This is basically the same, but better. :)

Reply:
While I definitely agree it's simpler from an implementer's point of view, I am struggling to see how this is better from a pure e2e perspective because we don't have any guarantees that the webapp will do its job to the fullest. I think this sort of more advanced cases would be nice to cover with something like [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) which would essentially let us intercept calls to the native APIs. But I also agree on being pragmatic, maybe something to consider for the future.